nexus/test-utils/src/background.rs

```diff
@@ -337,3 +375,68 @@ pub async fn run_replacement_tasks_to_completion(
     .await
     .unwrap();
 }
+
+pub async fn wait_tuf_artifact_replication_step(
+    internal_client: &ClientTestContext,
+) -> TufArtifactReplicationStatus {
+    let last_background_task =
+        wait_background_task(&internal_client, "tuf_artifact_replication")
+            .await;
+
+    let LastResult::Completed(last_result_completed) =
+        last_background_task.last
+    else {
+        panic!(
+            "unexpected {:?} returned from tuf_artifact_replication task",
+            last_background_task.last,
+        );
+    };
+
+    let status = serde_json::from_value::<TufArtifactReplicationStatus>(
+        last_result_completed.details,
+    )
+    .unwrap();
+    assert_eq!(status.requests_err, 0);
+    status
+}
+
+pub async fn run_tuf_artifact_replication_step(
+    internal_client: &ClientTestContext,
+) -> TufArtifactReplicationStatus {
+    let last_background_task =
+        activate_background_task(&internal_client, "tuf_artifact_replication")
+            .await;
+
+    let LastResult::Completed(last_result_completed) =
+        last_background_task.last
+    else {
+        panic!(
+            "unexpected {:?} returned from tuf_artifact_replication task",
+            last_background_task.last,
+        );
+    };
+
+    let status = serde_json::from_value::<TufArtifactReplicationStatus>(
+        last_result_completed.details,
+    )
+    .unwrap();
+    assert_eq!(status.requests_err, 0);
+    status
+}
+
+/// Run the `tuf_artifact_replication` task until the status indicates the task
+/// has stabilized: no outstanding requests, and no local repositories. Panics
+/// if the status does not change between runs.
+pub async fn run_tuf_artifact_replication_to_completion(
+    internal_client: &ClientTestContext,
+) {
+    let mut status = run_tuf_artifact_replication_step(internal_client).await;
+    while status.requests_outstanding > 0 || status.local_repos > 0 {
+        let new_status =
+            run_tuf_artifact_replication_step(internal_client).await;
+        if new_status == status {
+            panic!("TUF artifact replication stalled: {new_status:?}");
+        }
```

Comment:
This background task involves replicating to a random selection of sleds. If we pick the same set twice, would we stall, and trigger this panic?

Reply:
I don't think it's possible to pick the same set twice, unless errors are occurring, and we panic within these test functions if any errors are recorded by the task.